### PR TITLE
[#935] supporting application/json within multipart/form-data request body

### LINKED ISSUE #935: multipart/form-data content validation
### Description

I'm trying to have some validation on a REST route accepting a multipart/form-data content. This multipart is composed of 2 contents:

* a ZIP file named `stack_zip`  (application/zip)
* a form-data named `variables` (application/json)

multipart

### Expected behaviour

The `variables` content should be validated against a schema defined in openapi.yaml.

### Actual behaviour

The `variables` content is received by connexion as a raw string even if the `Content-Type` is set to `application/json`.

It also seems that Werkzeug is:

* keeping the headers when parsing the a File object  (application/zip)
* removing the headers when parsing any non-file object

### Steps to reproduce

OpenAPI schema.

```yaml

  /zones/{zoneId}/deploy_stack:
    post:
      tags:
      - Stack
      summary: Deploy a Docker Stack
      operationId: api.v1.view.deploy_stack
      responses:
        200:
          description: Successful response - Docker stack deployed.
        404:
          description: The specified `Zone` does not exist.
      requestBody:
        content:
          multipart/form-data:
            schema:
              type: object
              properties:
                variables:
                  type: object
                  properties:
                    stack_name:
                      type: string
                stack_zip:
                  type: string
                  format: binary
            encoding:
              stack_zip:
                contentType: application/zip
              variables:
                contentType: application/json

```

Client command:

```bash

curl -i -X POST 'http://localhost:9090/v1/zones/bo/deploy_stack'           -H 'Content-type:multipart/form-data'           -F 'stack_zip=@artifacts.zip;type=application/zip'           -F 'variables={"stak_name": "bar"}; type=application/json'
```

### Additional info:

 I tested this feature against the connexion 2.0 main branch as well as the branch from https://github.com/zalando/connexion/pull/760

I guess it would be nice if the library could parse the encoding section in the YAML specs and try to convert the form in a JSON object in the content-type is set to application/json. I guess I could start by tryiing to modify the code in #760 and add support for encoding.

### PR DESCRIPTION
**Pull request by ddurham2**
adding support to automatically decode json when a multipart/form-data request body contains a field with an application/json content-type

Fixes #935.

Changes proposed in this pull request:

Fills in a decode step where there's already a TODO (I think) saying it needs to be done for a variety of types.

### CODE REVIEW

**Comment by fffan64** [comment]
It solves my issue, any idea when it will be merged ?
Thank you :)

File: connexion/decorators/uri_parsing.py [hunk 1]
```diff
@@ -174,6 +175,8 @@ def resolve_form(self, form_data):
                 self._resolve_param_duplicates(form_data[k], encoding, 'form')
             if defn and defn["type"] == "array":
                 form_data[k] = self._split(form_data[k], encoding, 'form')
+            elif encoding.get('contentType', None) == 'application/json':
```
[the remark this answers is not in the document]
Comment (fffan64) [reply connexion/decorators/uri_parsing.py]: doesn't change that much for me, it is just the default val

**Comment by fffan64** [comment]
> adding support to automatically decode json when a multipart/form-data request body contains a field with an application/json content-type
> 
> Fixes #935.
> 
> Changes proposed in this pull request:
> 
> Fills in a decode step where there's already a TODO (I think) saying it needs to be done for a variety of types.

I forked your modif because i need it in one of my project quickly, thank you

File: connexion/decorators/uri_parsing.py [hunk 2]
```diff
@@ -2,6 +2,7 @@
 import abc
 import functools
 import logging
+import json
```
Comment (kornicameister) [inline connexion/decorators/uri_parsing.py]:
```suggestion
from .. import utils
```

File: connexion/decorators/uri_parsing.py (hunk 1 above)
Comment (kornicameister) [inline connexion/decorators/uri_parsing.py]:
```suggestion
            elif 'contentType' in encoding && utils.all_json([encoding.get['contentType'])]:
```

It is not perfect, the point is that you are reusing utility from connexion.
How do you feel about that?

**Comment by zhrkevin** [comment]
`
import json
if encoding.get('contentType', None) == "application/json":
    try:
        form_data[k] = json.loads(form_data[k])
    except Exception:
        pass
`
How do you feel about this?

**Comment by kornicameister** [comment]
@zhrkevin sorry for late response. It seems to me that me and connexion will part our ways. I am interested more in `aiohttp` support which seems to be 2nd class citizen here. That made ma start [axion](https://github.com/kornicameister/axion). If you or anyone wants to build typesafe all the way API first framework with async in mind...I will gladly accept any help 🤗

**Comment by timonbimon** [comment]
is connexion still maintained in general?

**Comment by kornicameister** [comment]
@timonbimon honestly speaking I don't know. It's very painful to say, but all signs says that active maintenance is not so much of a priority for Zalando at the moment. Well, I am not Zalando employee, just observing the situation.

**Comment by hjacobs** [comment]
@timonbimon @kornicameister sorry for the late answer: I'm one of the maintainers and currently don't have Connexion much on my radar. We have an internal discussion on how we can get more people involved and help maintain Connexion.

**Comment by hjacobs** [comment]
In order to move forward with this PR: I would at least like to see a proper unit test.

**Comment by kornicameister** [comment]
@hjacobs with all due respect and I am saying this with all best intentions in my heart. I'd see connexion progress and be maintained with the love it needs. It's just that really requires some sort of revamp in quite a few places. I have noticed that a lot of protocols a core abstract classes make are not met and it is done by either `aiohttp`, `flask` or, perhaps, even both implementations.

**Comment by ddurham2** [comment]
bump?

**Comment by ddurham2** [comment]
Unit test added. Works with new changes, fails without them.

**Comment by ddurham2** [comment]
buhbump?

**Comment by crertel-axios** [comment]
bomp

**Comment by hjacobs** [comment]
:+1:

**Pull request merged**